### PR TITLE
gitlib: allow using ssh remote url

### PR DESCRIPTION
**Pull request by nikhita**
Fixes https://github.com/kubernetes/release/issues/730

The remote url when using:

- https: `https://github.com/kubernetes/release.git`
- ssh: `git@github.com:kubernetes/release.git`

For ssh, there is no forward slash before `kubernetes`. So remove it
from `expectedRemote` so that it can match both https and ssh.

If string constants are used for regular expressions with awk, dynamic
regex is used and the string is scanned twice.
Ref: https://www.gnu.org/software/gawk/manual/html_node/Computed-Regexps.html

So add extra back slashes to actually escape the backslash for matching
the bracket in "fetch", and don't escape the forward slash as it isn't
needed (only back slashes need to be escaped).

Note: earlier, we weren't using string constants so the extra backslashes weren't
needed. Ref: https://github.com/kubernetes/release/commit/636269f29eb1caf416537c23cb6a6f9fb7022cc5#diff-700d6995a68e6e3a2be69788965225ee

/assign @hoegaarden @idealhack

### CODE REVIEW

File: lib/gitlib.sh [hunk 1]
```diff
@@ -299,8 +299,7 @@ gitlib::repo_state () {
     echo "${expectedRemote}" \
       | awk -F '[:/]' '{ print $(NF-1) "/" $NF }'
   )"
-  # make sure all '/' in the remote are escaped
-  remotePattern="${expectedRemote//\//\\\/}(.git)* \(fetch\)$"
+  remotePattern="${expectedRemote}(.git)* \\\(fetch\\\)$"
```
Comment (nikhita) [inline lib/gitlib.sh]: I actually don't understand how it works with https in https://github.com/kubernetes/release/issues/730. So maybe I'm missing something here... :thinking: 

Without the extra backslashes, I get this locally:

```
[nikhita@nikhita-dev release]$ git remote -v | awk "-vP=${remotePattern}" '$0 ~ P {print $1}'
awk: warning: escape sequence `\(' treated as plain `('
awk: warning: escape sequence `\)' treated as plain `)'
``

Comment (hoegaarden) [reply lib/gitlib.sh]: TL;DR: I think we should not use `awk` but `grep ... | cut ...` instead.

I did some tests [over there](https://gist.github.com/hoegaarden/d6d28ba3c6f4a89464b602ebf22baf10) (on `debian:sid` with `original-awk`, `gawk`, and `mawk` installed) and those seem to show:
- different `awk` implementations behave very different
- `grep -E ...` does not care if `/` are escaped or not

Therefore I opt for:
- `local expectedRemote="${RELEASE_TOOL_REPO:-[:/]kubernetes/release}"`
- `local remotePattern="${expectedRemote}(.git)* \(fetch\)$"`
- `local remote=$( git -C "$TOOL_ROOT" remote -v | grep -E "$remotePattern" | cut -f1 )`

Also, as side notes:
- https://github.com/kubernetes/release/blob/124d66d8967f7149c72984448987424ff60dbf41/lib/gitlib.sh#L298-L301 can be removed, as `remotePattern` will be set unconditionally on the next line
- I am confident that GNU grep and BSD grep behave differently too, but I don't care to support BSD grep here
- I am not sure, what we should do when multiple remotes match, but I don't care too much either as this whole thing should go away anyway soonish hopefully 🤞

Comment (hoegaarden) [reply lib/gitlib.sh]: @idealhack @nikhita
🔥 🐚 🔥

**Comment by justaugustus** [comment]
I think I'm fine with @hoegaarden's suggestion, but just dropping by to put an explicit hold on this. I want to make sure there's nothing shifting in the tooling until we close the books on the 1.15 release cycle.

/hold

File: lib/gitlib.sh (hunk 1 above)
Comment (nikhita) [reply lib/gitlib.sh]: @hoegaarden thanks, updated :fire:

**Comment by hoegaarden** [comment]
As 1.15 is out
/hold cancel

Thanks @nikhita 
/lgtm

**Pull request merged**